### ISSUE TITLE
Native Wayland support

### ISSUE THREAD

**Issue by PonasKovas** [PonasKovas/rlaunch#9]

**Comment by PonasKovas** [comment]
Oh boy

**Comment by felipec** [comment]
For me rlaunch just works on Xorg, but I'm trying Hyprland and although it launches (probably due to Xwayland), it doesn't receive focus.

Is there a way to at least fake the Xorg behavior through Xwayland?

**Comment by PonasKovas** [comment]
Hello,

I do not use rlaunch and have not used it for a long time now, not in a position to make patches at this time unfortunately. I should probably make this clear in the readme.

So unfortunately I can't fix your issue, and honestly I would probably recommend using a different, better supported tool.

However, if you're not scared of playing around with the code you may be able to fix this issue and I can provide some guidance.

The keyboard grabbing (which I think is pretty much unsupported in wayland, thats why it doesnt work) is [implemented here](https://github.com/PonasKovas/rlaunch/blob/master/src/x11.rs#L147) and you may have some luck if you also add a [`XSetInputFocus` call](https://tronche.com/gui/x/xlib/input/XSetInputFocus.html). I can't guarantee that this will work, but it might, and if you're willing to try to tackle this I will help you as much as I can. I will also accept PRs.

**Comment by felipec** [comment]
Interesting.

@PonasKovas what launcher do you use instead of rlaunch?

**Comment by PonasKovas** [comment]
I personally haved opted in for a vanilla KDE experience and just use the
start button to launch things hahahah... The days of ricing and tiling wms
are over for me

**Comment by felipec** [comment]
Weird. I'm not using a tiling WM nor am I big on ricing.

I use Xfce and rlaunch is a simple and efficient launcher for it.

**Comment by PonasKovas** [comment]
Yeah, I meant in the sense that I'm not tinkering with my computer as much anymore and I prefer to just leave things as they are if they work.

Another big reason is that my current computer is much more performant. Back when I made `rlaunch`, I had an old laptop and it actually made a huge difference in usability, because other launchers like `dmenu` that I tried using would take sometimes up to 20 seconds to start, while `rlaunch` would start pretty much instantaneously.

Now default things are fast enough already.

**Comment by felipec** [comment]
I tried `rlaunch` on GNOME and it actually worked correctly, same on KDE. It seems the behavior of Xwayland depends on the compositor, and so far the only one that doesn't work for me is Hyprland.

Weird.

**Comment by PonasKovas** [comment]
Seems like classic wayland stuff.
